android-wave-recorder/src/main/java/com/github/squti/androidwaverecorder/WaveRecorder.kt

```diff
@@ -113,38 +129,54 @@ class WaveRecorder(private var filePath: String) {
                 it(RecorderState.RECORDING)
             }
 
-            GlobalScope.launch(Dispatchers.IO) {
+            GlobalScope.launch(Dispatchers.Main) {
                 writeAudioDataToStorage()
             }
         }
     }
 
-    private suspend fun writeAudioDataToStorage() {
+    private suspend fun writeAudioDataToStorage() = withContext(Dispatchers.IO) {
         val bufferSize = AudioRecord.getMinBufferSize(
             waveConfig.sampleRate,
             waveConfig.channels,
             waveConfig.audioEncoding
         )
         val data = ByteArray(bufferSize)
-        val file = File(filePath)
-        val outputStream = file.outputStream()
+        val outputStream: OutputStream
+//        var length: Long
+        if (filePath != null) {
+            val file = File(filePath!!)
```

Comment:
You can use `filePath?.let { path -> ... }  ?: run { ... }` instead of if else to avoid assertion